src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp

```diff
@@ -1554,6 +1554,7 @@ CTranslatorDXLToPlStmt::TranslateDXLTvf(
 	List *target_list = TranslateDXLProjList(
 		project_list_dxlnode, &base_table_context, nullptr, output_context);
 
+	target_list = gpdb::ProcessRecordFuncTargetList(((FuncExpr *) rtfunc->funcexpr)->funcid, target_list);
```

Comment:
pls use the m_md_accessor, which may the cached.
```
// convert func_exp -> funcid to FuncMdId(), just create a `CMDIdGPDB`
const IMDFunction *md_func =
					md_accessor->RetrieveFunc(tvf_node->FuncMdId());

IMDFunction->GetResultTypeMdid() // check the RECORDID 

List *arg_types = gpdb::GetFuncArgTypes(funcexpr->funcid); 
... do the logic 
```

Reply:
oh, i guess we can direct get the function ret type?

```
IMDId *  func_rtype= tvf_node->ReturnTypeMdId();
... then check the func_rtype is the recordid ?
``

Reply:
Good suggestion, I will try it.

Reply:
I have tried in this way, but it will error report 'ERROR:  cache lookup failed for function 2249 (gpdbwrappers.cpp:1023)', which means function has not loaded into FuncMd cache. In my opinion, to preloading this FuncMd to cache maybe more has cost than using ((FuncExpr *) rtfunc->funcexpr)->funcid?